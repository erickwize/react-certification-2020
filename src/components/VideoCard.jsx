import React from 'react';
import styled from 'styled-components';

const VideoCard = ({ data }) => {
  const formattedDate = new Date(data.snippet.publishedAt);
  const ye = new Intl.DateTimeFormat('en', { year: 'numeric' }).format(formattedDate);
  const mo = new Intl.DateTimeFormat('en', { month: 'short' }).format(formattedDate);
  const da = new Intl.DateTimeFormat('en', { day: '2-digit' }).format(formattedDate);

  return (
    <Card data-testid={data.etag}>
      <header>
        <img src={data.snippet.thumbnails.medium.url} alt={data.snippet.title} />
        <div>{data.snippet.title}</div>
      </header>
      {data.snippet.description !== '' ? (
        <div>{data.snippet.description}</div>
      ) : (
        <div>{`${data.snippet.channelTitle} - ${da} ${mo}, ${ye}`}</div>
      )}
    </Card>
  );
};

export default VideoCard;

const Card = styled.div`
  overflow: hidden;
  border-radius: 5px;
  box-shadow: 0px 0px 15px -5px rgba(0, 0, 0, 0.75);

  header {
    position: relative;
    width: 100%;
    max-height: 200px;
    overflow: hidden;
    cursor: pointer;
    &:hover {
      img {
        width: 175%;
        object-position: -7.5em -2em;
      }
    }

    img {
      width: 150%;
      transition: 0.2s;
      object-fit: cover;
      object-position: -5em -1em;
    }

    div {
      position: absolute;
      bottom: 0;
      padding: 3%;
      background-image: linear-gradient(0deg, #000 0%, transparent 100%);
      width: 100%;
      color: white;
      text-shadow: #000 0px 0px 5px;
    }
  }

  & > div {
    padding: 3%;
    font-size: 0.7em;
  }
`;
