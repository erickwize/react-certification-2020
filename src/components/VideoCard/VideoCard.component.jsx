import React from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';

const VideoCardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  background: #ffffff;
  color: #000000;
  border-radius: 5px;
  width: 100%;
  height: 345px;
  align-self: center;
  padding: 12px;
  border: 1px solid #e4e4e4;
`;

const VideoCardImage = styled.img`
  width: 100%;
  height: 140px;
  object-fit: cover;
  border-radius: 5px;
`;

const VideoCardTitle = styled.h2`
  font-weight: 900;
  font-size: 16px;
  // word-break: break-all;
`;

const VideoCardDescription = styled.p`
  font-weight: 400;
  font-size: 14px;
  margin: 0;
  word-break: break-all;
  
  text-overflow: ellipsis;
  word-wrap: break-word;
  overflow: hidden;
  max-height: 5.4em;
  line-height: 1.8em;
`;

function VideoCard({ video }) {
  const { id, title, description, thumbnail } = video;
  return (
    <Link to={`/video/${id}`}>
      <VideoCardWrapper>
        <VideoCardImage src={thumbnail} alt="video thumbnail" />
        <VideoCardTitle>{title}</VideoCardTitle>
        <VideoCardDescription>{description}</VideoCardDescription>
      </VideoCardWrapper>
    </Link>
  );
}

export default VideoCard;
