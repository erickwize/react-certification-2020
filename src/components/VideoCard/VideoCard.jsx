import React from 'react';
import { Link } from 'react-router-dom';
import VideoBox from './VideoCard.styled';

function VideoCard({ data, videoList }) {
  const { title, description } = data?.snippet;
  const source = data?.snippet.thumbnails.default.url;
  const videoId = data?.id?.videoId;
  return (
    <VideoBox>
      <Link
        to={{
          pathname: `/video/${videoId}`,
          data: { data, videoList },
        }}
      >
        <img src={source} alt={title} />
        <h3>{title}</h3>
        <p data-testid="paragraph">{description}</p>
      </Link>
    </VideoBox>
  );
}

export default VideoCard;
