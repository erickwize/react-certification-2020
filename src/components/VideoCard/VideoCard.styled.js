import styled from 'styled-components';

const VideoBox = styled.div`
  height: 330px;
  width: 330px;
  border-radius: 10px;
  border: 1px solid #a0aec0;
  margin: 10px;
  cursor: pointer;
  position: relative;
  transform: none;
  background: ${({ theme }) => theme.cardTheme};

  :hover {
    transform: scale(1.03) translateZ(0px);
  }
`;

const VideoImg = styled.img`
  width: 100%;
  height: 150px;
  object-fit: cover;
  object-position: center center;
  border-top-left-radius: 10px;
  border-top-right-radius: 10px;
  position: relative;
`;

const VideoTitle = styled.h3`
  margin: 0 1rem;
  text-align: center;
  margin-top: 0px;
  font-size: 15px;
`;

const VideoDescription = styled.p`
  color: ${({ theme }) => theme.cardText};
  font-size: 13px;
  text-align: start;
  padding: 0px 5px 0px 10px;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-line-clamp: 5;
  -webkit-box-orient: vertical;
  position: relative;
`;

export { VideoBox, VideoImg, VideoTitle, VideoDescription };
