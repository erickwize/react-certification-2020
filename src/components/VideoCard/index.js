import React from 'react'
import styled from 'styled-components'

const VideoCardStyled = styled.div`
	width: 18%;
	margin: 20px 10px;
	border-radius: 5px;
	overflow: hidden;
	@media only screen and (max-width: 1400px) {
		width: 23%;
	}
	@media only screen and (max-width: 1100px) {
		width: 30%;
	}
	@media only screen and (max-width: 768px) {
		width: 45%;
	}
	@media only screen and (max-width: 475px) {
		width: 100%;
	}
`
const VideoThumbnail = styled.img`
	width: 100%;
	height: 120px;
	object-fit: cover;
	display: block;
`
const VideoInfo = styled.div``

const VideoTitle = styled.span`
	font-size: 14px;
	font-weight: 800;
	color: #444;
	padding: 5px;
	display: block;
`
const VideoDescription = styled.p`
	color: #888;
	font-size: 12px;
	font-style: italic;
	margin: 0;
	padding: 0 5px;
`

export default function VideoCard({ item }) {
	return (
		<VideoCardStyled>
			<VideoThumbnail src={item.snippet.thumbnails.default.url} alt="thumbnail" />
			<VideoInfo>
				<VideoTitle>{item.snippet.title}</VideoTitle>
				<VideoDescription>{item.snippet.description}</VideoDescription>
			</VideoInfo>
		</VideoCardStyled>
	)
}
