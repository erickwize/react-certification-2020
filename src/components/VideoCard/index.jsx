import React from 'react';

const VideoCard = (props) => (
  <div id="{props.key}" className="max-w-sm rounded overflow-hidden shadow-lg">
    <img src={props.urlImage} alt="" className="w-full" />
    <div className="px-2 py-4">
      <div className="font-bold text-black text-left font-sans mb-2">{props.title}</div>
      <div className="text-sm text-left font-sans">{props.description}</div>
      <div className="flex justify-between bg-gray-400">
        <div className="flex-shrink md:flex-shrink-0 text-xs self-start p-1">
          {props.publishedDate}
        </div>
        <div className="md:flex-shrink-0 bg-red-500 font-serif text-xs italic text-white p-1 self-end">
          {props.channel}
        </div>
      </div>
    </div>
  </div>
);

export default VideoCard;
