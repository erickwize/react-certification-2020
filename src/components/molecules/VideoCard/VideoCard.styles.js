import styled from 'styled-components';

export const CardContainer = styled.div`
  display: block;
  flex-direction: column;
  height: fit-content;
  width: 320px;
  height: 320px;
  background-color: rgba(185, 134, 217, 0.2);
  border-radius: 6px;
  margin-bottom: 30px;
  overflow: hidden;
  box-shadow: rgba(50, 44, 78, 0.15) 0px 5px 15px 0px;
  &:hover {
    position: 'absolute';
    opacity: 0.6;
    box-shadow: rgba(254, 223, 3, 0.12) 0px 2px 4px 0px,
      rgba(254, 223, 3) 0px 2px 16px 0px;
  }
  @media screen and (max-width: 812px) {
    width: 95%;
    max-width: 320px;
    height: 320px;
  }
  cursor: not-allowed;
`;

// box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;

export const TextContainer = styled.div`
  margin: 0.5rem;
`;
