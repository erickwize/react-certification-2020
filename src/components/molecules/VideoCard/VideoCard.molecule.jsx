import React from 'react';
import { CardContainer, TextContainer } from './VideoCard.styles';
import { Thumbnail, VideoInformation } from '../../atoms';

function VideoCard({ videos }) {
  return (
    <CardContainer>
      <Thumbnail bg={<img src={videos.snippet.thumbnails.medium.url} />} />
      <TextContainer>
        <VideoInformation
          title={videos.snippet.title}
          description={videos.snippet.description}
        />
      </TextContainer>
    </CardContainer>
  );
}

export default VideoCard;
