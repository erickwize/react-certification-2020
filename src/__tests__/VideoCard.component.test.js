import React from 'react';
import renderer from 'react-test-renderer';
import VideoCard from '../components/VideoCard';

it('renders correctly', () => {
  const tree = renderer.create(<VideoCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
