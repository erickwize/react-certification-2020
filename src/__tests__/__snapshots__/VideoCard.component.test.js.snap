// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBwQ jqRnJH"
>
  <div
    className="sc-gsTCUz jzfVfp"
  />
  <h3
    className="sc-dlfnbm udFbv"
  />
  <p
    className="sc-hKgILt gbWVGw"
  />
</div>
`;
